src/main/java/io/strimzi/test/container/StrimziKafkaContainer.java

```diff
@@ -92,8 +93,15 @@ private StrimziKafkaContainer(CompletableFuture<String> imageName) {
 
     @Override
     protected void doStart() {
-        if (!imageNameProvider.isDone()) {
-            imageNameProvider.complete(KafkaVersionService.strimziTestContainerImageName(kafkaVersion));
+        if (!this.imageNameProvider.isDone()) {
+            this.imageNameProvider.complete(KafkaVersionService.strimziTestContainerImageName(this.kafkaVersion));
+        }
+        try {
+            if (this.useKraft && ((this.kafkaVersion != null && this.kafkaVersion.equals("2.8.1")) || this.imageNameProvider.get().contains("2.8.1"))) {
+                throw new UnsupportedKraftKafkaVersionException("Specified Kafka version " + this.kafkaVersion + " is not supported in KRaft mode.");
+            }
+        } catch (InterruptedException | ExecutionException e) {
+            LOGGER.error("Error occurred during retrieving of image name provider", e);
```

Comment:
I guess this should throw some other exception?

Reply:
This exception is possibly thrown when I invoke `this.imageNameProvider.get().` That's why it is needed here

Reply:
Sure, but why you just log it and ignore it? It clearly suggests the check didn't complete and there might be some issue.

Reply:
Ah, I see your point. I can add line, which will printStackTrace (`e.printStackTrace();`)